source/extensions/filters/http/grpc_stats/grpc_stats_filter.cc

```diff
@@ -278,6 +306,8 @@ class GrpcStatsFilter : public Http::PassThroughFilter {
   Grpc::FrameInspector response_counter_;
   Upstream::ClusterInfoConstSharedPtr cluster_;
   absl::optional<Grpc::Context::RequestStatNames> request_names_;
+  absl::optional<Grpc::Context::RequestStatNames> request_names_without_dots_{};
+  std::unique_ptr<std::string> service_name_without_dots_;
```

Comment:
You don't need `service_name_without_dots_` as it is already in `request_names_without_dots_`. Do not use pointer of a string.

You should be able to just store dots replaced names in `request_names_` directly instead of adding these.

Reply:
Thanks for the advice. I made the change to store the strings in `request_names_` and updated the tests.